scheduler/starter/modules.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/twitter/scoot/saga"
+	"github.com/twitter/scoot/saga/sagalogs"
+	"github.com/twitter/scoot/scheduler/scheduler/config"
+	"github.com/twitter/scoot/scheduler/server"
+	"github.com/twitter/scoot/workerapi/client"
+)
+
+const DEFAULT_SOCKS_PORT = "50001"
+const DEFAULT_SOCKS_ADDR = "localhost:" + DEFAULT_SOCKS_PORT
+
+func GetWorkerRunnerServiceFn(workers client.WorkersClientJSONConfig, thriftTransportFactory thrift.TTransportFactory, binaryProtocolFactory thrift.TProtocolFactory) (func(cluster.Node) runner.Service, error) {
+	var rf func(cluster.Node) runner.Service
+	var err error
+	if workers.Type == "socks" {
+		rf, err = GetSocksWorker()
+	} else {
+		rf, err = client.NewWorkerThriftClient(thriftTransportFactory, binaryProtocolFactory, common.DefaultClientTimeout, workers)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error creating worker thrift client (runner.Service). %s", err)
+	}
+
+	return rf, nil
+}
+
+func GetSocksWorker() (server.RunnerFactory, error) {
+	transportFactory, err := NewTTransportFactory()
+	if err != nil {
+		return nil, err
+	}
+	protocolFactory := thrift.NewTBinaryProtocolFactoryDefault()
+
+	wf := func(node cluster.Node) runner.Service {
+		di := dialer.NewSimpleDialer(transportFactory, protocolFactory, common.DefaultClientTimeout)
+		cl, _ := client.NewSimpleClient(di, string(node.Id()))
+		return runners.NewPollingService(cl, cl, 250*time.Millisecond)
+	}
+
+	return wf, nil
+}
+
+func GetProxyAddr() string {
+	port := os.Getenv("SOCKS_PORT")
+	if port != "" {
+		return "localhost:" + port
+	}
+	return ""
+}
+
+func GetProxyDialer(proxyAddr string) (proxy.Dialer, error) {
+	return proxy.SOCKS5("tcp", proxyAddr, nil, proxy.Direct)
+}
+
+func NewTTransportFactory() (thrift.TTransportFactory, error) {
+	addr := GetProxyAddr()
+	if addr == "" {
+		addr = DEFAULT_SOCKS_ADDR
+	}
+	dialer, err := GetProxyDialer(addr)
+	if err != nil {
+		return nil, err
+	}
+	return &socksTransportFactory{dialer}, nil
+}
+
+type socksTransportFactory struct {
+	dialer proxy.Dialer
+}
+
+func (f *socksTransportFactory) GetTransport(trans thrift.TTransport) thrift.TTransport {
+	// If this is a SOCKS-able connection, then return a TTransport that uses SOCKS.
+	// If it's not, return trans unmodified.
+	tsock, ok := trans.(*thrift.TSocket)
+	if !ok {
+		// Don't know how to wrap anything other than a TSocket
+		return trans
+	}
+	if tsock.IsOpen() {
+		// Transport Factory should be called before Open
+		return trans
+	}
+	addr, ok := tsock.Addr().(*net.TCPAddr)
+	if !ok {
+		return trans
+	}
+	ip := addr.IP.To4()
+	if ip == nil {
+		return trans
+	}
+
+	// An Address is Socks-able if it starts with 10, because Twitter's production network lives in
+	// 10/8 in IPv4 space.
+	if ip[0] != 10 {
+		return trans
+	}
+
+	if f.dialer == nil {
+		// We want to use SOCKS, but can't, because we have no dialer
+		log.Info("Aurora address specified, but cannot use SOCKS proxy because SOCKS_PORT is unset", addr.String())
```

Comment:
oss shouldn't reference "Aurora". not sure what that implies as to what this logic was supposed to be doing internally

Reply:
modified the comment